homeassistant/components/solaredge/sensor.py

```diff
@@ -312,11 +315,11 @@ def __init__(
     ) -> None:
         """Initialize the sensor."""
         super().__init__(data_service.coordinator)
-        self.platform_name = platform_name
         self.entity_description = description
         self.data_service = data_service
-
-        self._attr_name = f"{platform_name} ({description.name})"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, data_service.site_id)}, name=platform_name
```

Comment:
Platform name is the config entry title, which means that setting a name here isn't needed and the passing of it can be cleaned up.

../Frenck

Reply:
I tried that, but all the tests failed, so I'm not 100% sure what the best way is here

Reply:
I now know why, it wasn't a primary device with only identifiers :)